OpenRA.Mods.Common/Traits/CreatesShroud.cs

```diff
@@ -37,5 +43,16 @@ protected override void AddCellsToPlayerShroud(Actor self, Player p, PPos[] uv)
 		}
 
 		protected override void RemoveCellsFromPlayerShroud(Actor self, Player p) { p.Shroud.RemoveSource(this); }
+
+		public override WDist Range
+		{
+			get
+			{
+				var revealsShroudModifier = self.TraitsImplementing<ICreatesShroudModifier>().Select(x => x.GetCreatesShroudModifier());
+				var range = Util.ApplyPercentageModifiers(Info.Range.Length, revealsShroudModifier.ToArray());
+
+				return CachedTraitDisabled ? WDist.Zero : new WDist(range);
```

Comment:
Can we make this an if check at the beginning?
```
if (CachedTraitDisabled)
	return WDist.Zero;

[...]
```
That way we don't have to do a lot of unnecessary calculation.